### DISCUSSION TITLE
Preventing Screen Recording or Capture in Fullscreen Mode

### DISCUSSION THREAD

**Discussion by programmer1zero1** [software-mansion/react-native-screens#2331] [General]
Is there anyone who can provide guidance on preventing screen recording or capture in fullscreen mode for both iOS and Android? On Android, I tried implementing **FLAG_SECURE**, but it only affects the entire app and not specifically fullscreen mode. What could be the solution?

**Comment by tboba** [comment]
Hi @programmer1zero1, react-native-screens doesn't give such feature to prevent screen recording. However, you can check other libraries that supports this functionality, e.g. `react-native-capture-protection` - https://github.com/wn-na/react-native-capture-protection.

For now I'll close this discussion, but for any other questions don't hesitate to re-open it.
Cheers!

**Reply by programmer1zero1** [reply]
@tboba 
"Hey, good to see a response. I’ve been able to handle screen recording by exiting full-screen mode when recording starts. However, without closing the video, this library[react-native-capture-protection](https://github.com/wn-na/react-native-capture-protection) doesn't seem to prevent screen recording in  [react-native-video](https://www.npmjs.com/package/react-native-video) full-screen mode."

**Reply by tboba** [reply]
@programmer1zero1 I understand. Nonetheless, this seems not to be a responsibility of react-native-screens to somehow detect full-screen mode, sorry 😔.
Could you create an issue in react-native-video then?

**Reply by programmer1zero1** [reply]
@tboba Thank you for the clarification and guidance! I understand that detecting full-screen mode is not within the scope of react-native-screens. I will go ahead and create an issue in the react-native-video repository to address this. I appreciate your help! 😊
